x-pack/plugins/observability_solution/entities_data_access/server/services/index_pattern_service.test.ts

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { savedObjectsClientMock } from '@kbn/core-saved-objects-api-server-mocks';
+import { createIndexPatternService } from './index_pattern_service';
+
+describe('index_pattern_service', () => {
+  const indexPatternService = createIndexPatternService();
+
+  describe('#indexPattern', () => {
+    it('returns wide index pattern when all datasets and version specified', () => {
+      let pattern = indexPatternService.indexPattern();
+      expect(pattern).toEqual('.entities.*.*.*');
+
+      pattern = indexPatternService.indexPattern({ datasets: 'all', schemaVersion: 'all' });
+      expect(pattern).toEqual('.entities.*.*.*');
+    });
+
+    it('returns dataset-scoped pattern when specified', () => {
+      let pattern = indexPatternService.indexPattern({ datasets: 'history' });
+      expect(pattern).toEqual('.entities.*.history.*');
+
+      pattern = indexPatternService.indexPattern({ datasets: 'latest' });
+      expect(pattern).toEqual('.entities.*.latest.*');
+    });
+
+    it('returns version-scoped pattern when specified', () => {
+      let pattern = indexPatternService.indexPattern({ datasets: 'history', schemaVersion: 'v1' });
+      expect(pattern).toEqual('.entities.v1.history.*');
+
+      pattern = indexPatternService.indexPattern({
+        datasets: 'latest',
+        schemaVersion: ['v1', 'v2'],
+      });
+      expect(pattern).toEqual('.entities.v1.latest.*,.entities.v2.latest.*');
+    });
+  });
+
+  describe('#indexPatternByDefinitionId', () => {
+    it('returns wide index pattern when all datasets/version specified', () => {
+      let pattern = indexPatternService.indexPatternByDefinitionId('my_definition_id');
+      expect(pattern).toEqual(
+        '.entities.*.latest.my_definition_id,.entities.*.history.my_definition_id.*'
+      );
+
+      pattern = indexPatternService.indexPatternByDefinitionId('my_definition_id', {
+        datasets: 'all',
+        schemaVersion: 'all',
+      });
+      expect(pattern).toEqual(
+        '.entities.*.latest.my_definition_id,.entities.*.history.my_definition_id.*'
```

Comment:
With the current index naming scheme we can’t have a unique index pattern that matches all datasets. Should we consider adding a trailing namespace like in the datastream naming scheme ? (in our case it could be the space in which the definition is installed, .entities.v1.{dataset}.{definition_id}-{namespace})